src/spherical_mesh.cc

```diff
@@ -1,33 +1,24 @@
 #include "flyft/spherical_mesh.h"
 
 #include <cmath>
+#include <exception>
 
 namespace flyft
 {
-
-SphericalMesh::SphericalMesh(double R, int shape)
-    : SphericalMesh(R,shape,0)
+SphericalMesh::SphericalMesh(double lower_bound, double upper_bound,int shape, BoundaryType lower_bc, BoundaryType upper_bc)
+    : Mesh(lower_bound, upper_bound, shape, lower_bc, upper_bc)
     {
+      validateBoundaryCondition();
     }
 
-SphericalMesh::SphericalMesh(double R,int shape, double origin)
-    : Mesh(R,shape,origin)
-    {
-    }
-
-SphericalMesh::SphericalMesh(int shape, double step)
-    :SphericalMesh(shape,step,0)
-    {
-    }
-
-SphericalMesh::SphericalMesh(int shape, double step, double origin)
-    : Mesh(shape, step, origin)
-    {
-    }
 
 std::shared_ptr<Mesh> SphericalMesh::slice(int start, int end) const
     {
-    return std::shared_ptr<Mesh>(new SphericalMesh(end-start,step_,lower_bound(start)));
+    return std::shared_ptr<Mesh>(new SphericalMesh(
+        lower_bound(start),
+        lower_bound(end),end-start, 
+        (start > 0)?BoundaryType::internal:lower_bc_, 
+        (start > 0)?BoundaryType::internal:upper_bc_));
```

Comment:
This change did not get applied correctly (importantly, the second ternary check was wrong):
```suggestion
        lower_bound(start),
        lower_bound(end),
        end-start, 
        (start > 0) ? BoundaryType::internal : lower_bc_, 
        (end > shape_) ? BoundaryType::internal : upper_bc_));
```